### ISSUE TITLE
Error with Latest PhpStorm EAP (Build #PS-212.3724.24, built on June 4, 2021)

### ISSUE THREAD

**Issue by s001911** [adelf/idea-php-dotenv-plugin#77]
PhpStorm version:
```
PhpStorm 2021.2 EAP
Build #PS-212.3724.24, built on June 4, 2021
PhpStorm EAP User
Expiration date: July 4, 2021
Runtime version: 11.0.11+9-b1481.1 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 5.12.9-zen1-1-zen
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 8
Registry: run.processes.with.pty=TRUE
Non-Bundled Plugins: String Manipulation (8.15.203.000.3), name.kropp.intellij.makefile (212.3724.3), de.espend.idea.php.annotation (8.0.0), fr.adrienbrault.idea.symfony2plugin (0.23.209), ru.adelf.idea.dotenv (2021.2)
Current Desktop: GNOME
```
Error:
```
java.lang.Throwable: Class com.intellij.docker.dockerFile.DockerFileType must be not requested from main classloader of Docker plugin
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:182)
	at com.intellij.ide.plugins.ClassLoaderConfiguratorKt$createPluginDependencyAndContentBasedScope$1.isDefinitelyAlienClass(ClassLoaderConfigurator.kt:467)
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClassInsideSelf(PluginClassLoader.java:358)
	at com.intellij.ide.plugins.cl.PluginClassLoader.tryLoadingClass(PluginClassLoader.java:275)
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.java:244)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at ru.adelf.idea.dotenv.docker.DockerfileVariablesProvider.acceptFile(DockerfileVariablesProvider.java:19)
	at ru.adelf.idea.dotenv.indexing.DotEnvKeyValuesIndex.lambda$getInputFilter$1(DotEnvKeyValuesIndex.java:65)
	at com.intellij.util.indexing.FileBasedIndexImpl.lambda$composeInputFilter$36(FileBasedIndexImpl.java:2001)
	at com.intellij.util.indexing.FileBasedIndexImpl.acceptsInput(FileBasedIndexImpl.java:1989)
	at com.intellij.util.indexing.FileBasedIndexImpl.acceptsInput(FileBasedIndexImpl.java:1874)
	at com.intellij.util.indexing.FileBasedIndexImpl.shouldIndexFile(FileBasedIndexImpl.java:1830)
	at com.intellij.util.indexing.UnindexedFilesFinder.lambda$getFileStatus$2(UnindexedFilesFinder.java:121)
	at com.intellij.openapi.fileTypes.impl.FileTypeManagerImpl.freezeFileTypeTemporarilyIn(FileTypeManagerImpl.java:631)
	at com.intellij.util.indexing.UnindexedFilesFinder.lambda$getFileStatus$3(UnindexedFilesFinder.java:103)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:853)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:61)
	at com.intellij.util.indexing.UnindexedFilesFinder.getFileStatus(UnindexedFilesFinder.java:58)
	at com.intellij.util.indexing.UnindexedFilesUpdater.lambda$collectIndexableFilesConcurrently$13(UnindexedFilesUpdater.java:432)
	at com.intellij.openapi.vfs.VfsUtilCore$1.visitFileEx(VfsUtilCore.java:278)
	at com.intellij.openapi.vfs.VfsUtilCore.visitChildrenRecursively(VfsUtilCore.java:294)
	at com.intellij.openapi.vfs.VfsUtilCore.iterateChildrenRecursively(VfsUtilCore.java:274)
	at com.intellij.openapi.vfs.VfsUtilCore.iterateChildrenRecursively(VfsUtilCore.java:267)
	at com.intellij.util.indexing.roots.IndexableFilesIterationMethods.iterateRoots(IndexableFilesIterationMethods.kt:31)
	at com.intellij.util.indexing.roots.IndexableSetContributorFilesIterator.iterateFiles(IndexableSetContributorFilesIterator.kt:50)
	at com.intellij.util.indexing.UnindexedFilesUpdater.lambda$collectIndexableFilesConcurrently$14(UnindexedFilesUpdater.java:448)
	at com.intellij.openapi.roots.impl.PushedFilePropertiesUpdaterImpl$6.run(PushedFilePropertiesUpdaterImpl.java:352)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:183)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:705)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:647)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:63)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:170)
	at com.intellij.openapi.roots.impl.PushedFilePropertiesUpdaterImpl.lambda$invokeConcurrentlyIfPossible$16(PushedFilePropertiesUpdaterImpl.java:359)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:216)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$200(BoundedTaskExecutor.java:27)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.execute(BoundedTaskExecutor.java:195)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:213)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:184)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by g011um** [comment]
A bit anecdotal, but having this plug-in enabled seems to cause issues with phpStorm -- autocomplete features for HTML no longer work (things like auto-adding "class", "id" attributes etc); can't type anything within single quoted strings in PHP code; auto-indent of code/HTML no longer works.  Disabling the env module fixed all of those issues for me.

I don't know if this is the fault of the env plug-in, or an issue with phpStorm itself and how it deals with having plug-ins that aren't supported enabled.

**Comment by IcyPalm** [comment]
Can confirm the same issue with WebStorm 2021.2 EAP

```
WebStorm 2021.2 EAP
Build #WS-212.3724.2, built on June 1, 2021
WebStorm EAP User
Expiration date: July 1, 2021
Runtime version: 11.0.11+9-b1481.1 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 10 10.0
GC: G1 Young Generation, G1 Old Generation
Memory: 750M
Cores: 4
Non-Bundled Plugins: com.ivanovych666.intellij.plugin.jsonsorter (1.0.0-beta.3.4), com.krylysov.nsisplugin (0.2.0), net.vektah.codeglance (1.5.4), ru.adelf.idea.dotenv (2021.2), PlantUML integration (5.4.0)
```

**Comment by adelf** [comment]
There are a lot of similar errors like that: https://youtrack.jetbrains.com/issues?q=main%20classloader%20Docker%20plugin

Seems, something is wrong with a new Docker plugin for 2021.2 versions.
https://youtrack.jetbrains.com/issue/DEXP-590427
https://youtrack.jetbrains.com/issue/IDEA-271231

**Comment by adelf** [comment]
A special update without Docker and JS support was released for 2021.2
